### PR TITLE
#158174587 test accounts 

### PR DESCRIPTION
**Pull request by matthewacha**
#### What does this PR do?
- Fixes unit tests in the directory hc/accounts/tests
- Adds new tests for untested components
 standards
#### Description of Task to be completed?
Implementing the following test cases
- test_check_token.py
  - test_redirects_already_logged_in
  - test_login_bad_token_redirects
- test_login.py
  - test_it_redirects_when_password_given
- test_profile.py
 - test_it_sends_set_password_link
   - Assert that the token is set
   - Assert that the email was sent and check email content
 - test_it_sends_report
   - Assert that the email was sent and check email content
 - test_it_adds_team_member
   - Assert the existence of the member emails
   - Assert that the email was sent and check email content
 - test_it_creates_and_revokes_api
 - test_it_updates_reports
-test_switch_team.py
 - test_it_switches
   - Assert the contents of r
 - test_it_checks_team_membership
   - Assert the expected error code
 - test_it_switches_to_own_team
   - Assert the expected response code
 - test_team_access_middleware.py
   - test_it_handles_missing_profile
     -Assert the new Profile objects count

#### How should this be manually tested?
After cloning the repo, run the following commands
- make
- python manage.py test
#### Any background context you want to provide?
This is necessary before any further code implementation is made to the code base
#### What are the relevant pivotal tracker stories?
[#158174587](https://www.pivotaltracker.com/n/projects/2177206)

### CODE REVIEW

**Comment by magicmarie** [comment]
Nice one on the docstrings. The rest of us should pick a leaf

File: hc/accounts/tests/test_check_token.py [hunk 1]
```diff
@@ -10,19 +10,30 @@ def setUp(self):
...
-    ### Login with a bad token and check that it redirects
-
-    ### Any other tests?
+    def test_redirects_already_logged_in(self):
+        """test that a logged in user redirects when they log in again""" 
+        #Login and test it redirects already logged in
+        self.client.post("/accounts/check_token/alice/secret-token/")
+        re = self.client.post("/accounts/check_token/alice/secret-token/")
+        self.assertEqual(re.status_code, 302)
+        self.assertRedirects(re, "/checks/")
```
Comment (EdwinKato) [inline hc/accounts/tests/test_check_token.py]: Use reverse for these urls `reverse("hc-checks")`

File: hc/accounts/tests/test_check_token.py [hunk 2]
```diff
@@ -10,19 +10,30 @@ def setUp(self):
...
+
+    def test_login_bad_token_redirects(self):
+        """test that login with bad token redirects back to login page"""
+        #Login with a bad token and check that it redirects
+        self.client.post("/accounts/login/", {"email":"bob@example.org"},follow=True)
+        response_bad_token = self.client.post("/accounts/check_token/bob/bad-token/")
+        self.assertRedirects(response_bad_token,"/accounts/login/")
```
Comment (EdwinKato) [inline hc/accounts/tests/test_check_token.py]: You can do without line 37. Leave a space `response_bad_token`

**Review by EdwinKato — CHANGES_REQUESTED** [review]

**Comment by EdwinKato** [comment]
@matthewacha resolve conflicts

**Review by EdwinKato — APPROVED** [review]

**Pull request merged**